source/SIL.AppBuilder.Portal.Frontend/src/lib/promises.ts

```diff
@@ -0,0 +1,13 @@
+export function timeoutablePromise(timeoutMs, promise) {
+  const timeout = new Promise((resolve, reject) => {
+    const id = setTimeout(() => {
+      clearTimeout(id);
+      reject(`Timed out after ${timeoutMs} ms.`);
```

Comment:
I don't think this needs translating. it *should* never happen and is really for our (the devs) benefit

Reply:
agree, just a suggestion invert the properties and give it a default 

timeoutablePromise(promise, timeoutMs=5000)

Reply:
if we use this again, we can give it a default :)